<a href="https://colab.research.google.com/github/michellssGUi/dataScience/blob/main/Wallmart_Melhor_Loja.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Análise Lojas Wallmart

---

# Introdução

## 1.1 - Objetivo do Projeto
> Fazer um levantamento do faturamento das lojas nos USA e apontar qual loja seria melhor para expandir seu tamaho.

## 1.2 Descrevendo os dados:
> Os dados apresentam as vendas semanais de 2010/02/05 à 2012/11/12 em umtotal de 45 lojas varejistas da rede Wallmart e também algumas métricas econômicas e meteorológicas por semana. 

A descrição destas características estão abaixo:

## Feriados Importantes

Holiday | Semanas
----|----
Super Bowl |	12-Feb-10, 11-Feb-11, 10-Feb-12
Labour Day | 10-Sep-10, 09-Sep-11, 07-Sep-12
Thanksgiving | 26-Nov-10, 25-Nov-11, 23-Nov-12
Cristmans | 31-Dec-10, 30-Dec-11, 28-Dec-12

## Index's das tabelas e suas descrições

Variável | Descrição
----|----
Store | Número da loja
Date | Semana de venda
Weekly_Sales | Vendas Nauela Semana
Hliday_Flag | Flag se é ou não semana com feriado (1 - Holiday Weeh e 0 - Non Holiday Week)
Temeratura | Temperatura do dia em °F
Fuel_Price | Preço do combustível na região da loja
CPI | Índice de preços ao consumidor
Unemployment | Taxa de desemprego


## 2 - Os dados
> Realizandoimportação das bibliotecas python as quais serão utilizadas durante o processo de limpeza, organização e análise dos dados.

In [411]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import scipy as scp
import seaborn as sns

### 2.1 - Importanto os dados
> Os dados se encontram em um aquivo .CSV e foram importados fazendo us da biplioteca "pandas".

In [412]:
df = pd.read_csv('Walmart.csv')
df

Store        Date  Weekly_Sales  Holiday_Flag  Temperature  Fuel_Price  \
0         1  05-02-2010    1643690.90             0        42.31       2.572   
1         1  12-02-2010    1641957.44             1        38.51       2.548   
2         1  19-02-2010    1611968.17             0        39.93       2.514   
3         1  26-02-2010    1409727.59             0        46.63       2.561   
4         1  05-03-2010    1554806.68             0        46.50       2.625   
...     ...         ...           ...           ...          ...         ...   
6430     45  28-09-2012     713173.95             0        64.88       3.997   
6431     45  05-10-2012     733455.07             0        64.89       3.985   
6432     45  12-10-2012     734464.36             0        54.47       4.000   
6433     45  19-10-2012     718125.53             0        56.47       3.969   
6434     45  26-10-2012     760281.43             0        58.85       3.882   

             CPI  Unemployment  
0     211.096358         8.106  
1     211.242170         8.106  
2     211.289143         8.106  
3     211.319643         8.106  
4     211.350143         8.106  
...          ...           ...  
6430  192.013558         8.684  
6431  192.170412         8.667  
6432  192.327265         8.667  
6433  192.330854         8.667  
6434  192.308899         8.667  

[6435 rows x 8 columns]

In [413]:
df = df.sort_values("Store")
df

Store        Date  Weekly_Sales  Holiday_Flag  Temperature  Fuel_Price  \
0         1  05-02-2010    1643690.90             0        42.31       2.572   
92        1  11-11-2011    1594938.89             0        59.11       3.297   
93        1  18-11-2011    1539483.70             0        62.25       3.308   
94        1  25-11-2011    2033320.66             1        60.14       3.236   
95        1  02-12-2011    1584083.95             0        48.91       3.172   
...     ...         ...           ...           ...          ...         ...   
6340     45  07-01-2011     680254.35             0        34.32       3.193   
6341     45  14-01-2011     654018.95             0        24.78       3.205   
6342     45  21-01-2011     644285.33             0        30.55       3.229   
6362     45  10-06-2011     814395.17             0        73.88       3.881   
6434     45  26-10-2012     760281.43             0        58.85       3.882   

             CPI  Unemployment  
0     211.096358         8.106  
92    217.998085         7.866  
93    218.220509         7.866  
94    218.467621         7.866  
95    218.714733         7.866  
...          ...           ...  
6340  182.598306         8.549  
6341  182.658578         8.549  
6342  182.919337         8.549  
6362  185.648692         8.521  
6434  192.308899         8.667  

[6435 rows x 8 columns]

## 2.2 - Limpeza e preparação dos dados
> Durante esse processo preparandos os dados para análise realizando a limpeza, transforamção e reorganização à medida que for necessário.

### 2.2.1 - Tratamento dos dados ausentes
> Dados ausentes são comuns em muitas aplicações de análise de dados. Um dos objetivos desse passo é deixar o trabalho o menos prolemático possível.


> Por padrão, "dropna()" descarta qualquer linha contendo um valor ausente. Devido a isso é importante avaliar se a perda dessa linhas não irá impactar diretamento na análise dos dados.

In [414]:
df = df.dropna()
df

Store        Date  Weekly_Sales  Holiday_Flag  Temperature  Fuel_Price  \
0         1  05-02-2010    1643690.90             0        42.31       2.572   
92        1  11-11-2011    1594938.89             0        59.11       3.297   
93        1  18-11-2011    1539483.70             0        62.25       3.308   
94        1  25-11-2011    2033320.66             1        60.14       3.236   
95        1  02-12-2011    1584083.95             0        48.91       3.172   
...     ...         ...           ...           ...          ...         ...   
6340     45  07-01-2011     680254.35             0        34.32       3.193   
6341     45  14-01-2011     654018.95             0        24.78       3.205   
6342     45  21-01-2011     644285.33             0        30.55       3.229   
6362     45  10-06-2011     814395.17             0        73.88       3.881   
6434     45  26-10-2012     760281.43             0        58.85       3.882   

             CPI  Unemployment  
0     211.096358         8.106  
92    217.998085         7.866  
93    218.220509         7.866  
94    218.467621         7.866  
95    218.714733         7.866  
...          ...           ...  
6340  182.598306         8.549  
6341  182.658578         8.549  
6342  182.919337         8.549  
6362  185.648692         8.521  
6434  192.308899         8.667  

[6435 rows x 8 columns]

> Nesse caso não hove perda de lihas, continuamos com as mesmas 6435 row's e 8 colums.

> OBS: Em caso de se perceber a perda demasiada de row's e por conseuência a perda de dados importantes, poderia-se fazer a substituição dessesdados por um alor consntante, pela média e ou mediana dos demais valores.

Segue exemplo:

> A funçao "fillna()" devolve um novo objeto reindexado com as condições que foram passadas... nesse caso retorna a media cos campos de valor 'NaN'.

In [415]:
df = df.fillna(df.mean())
df

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  """Entry point for launching an IPython kernel.


Store        Date  Weekly_Sales  Holiday_Flag  Temperature  Fuel_Price  \
0         1  05-02-2010    1643690.90             0        42.31       2.572   
92        1  11-11-2011    1594938.89             0        59.11       3.297   
93        1  18-11-2011    1539483.70             0        62.25       3.308   
94        1  25-11-2011    2033320.66             1        60.14       3.236   
95        1  02-12-2011    1584083.95             0        48.91       3.172   
...     ...         ...           ...           ...          ...         ...   
6340     45  07-01-2011     680254.35             0        34.32       3.193   
6341     45  14-01-2011     654018.95             0        24.78       3.205   
6342     45  21-01-2011     644285.33             0        30.55       3.229   
6362     45  10-06-2011     814395.17             0        73.88       3.881   
6434     45  26-10-2012     760281.43             0        58.85       3.882   

             CPI  Unemployment  
0     211.096358         8.106  
92    217.998085         7.866  
93    218.220509         7.866  
94    218.467621         7.866  
95    218.714733         7.866  
...          ...           ...  
6340  182.598306         8.549  
6341  182.658578         8.549  
6342  182.919337         8.549  
6362  185.648692         8.521  
6434  192.308899         8.667  

[6435 rows x 8 columns]

### 2.2.2 - Tratamento de dados duplicados
> Linhas duplicadaspodem ser encontradas em um DataFrame pr diversos motivos, por isso se faz necessário verificar sua existência e tata-los.

> Farei uso do método "duplicated()" para verificar a existência de dados dupicados.

In [416]:
df.duplicated()

0       False
92      False
93      False
94      False
95      False
        ...  
6340    False
6341    False
6342    False
6362    False
6434    False
Length: 6435, dtype: bool

In [417]:
df = df.drop_duplicates()
pd.DataFrame(df)

Store        Date  Weekly_Sales  Holiday_Flag  Temperature  Fuel_Price  \
0         1  05-02-2010    1643690.90             0        42.31       2.572   
92        1  11-11-2011    1594938.89             0        59.11       3.297   
93        1  18-11-2011    1539483.70             0        62.25       3.308   
94        1  25-11-2011    2033320.66             1        60.14       3.236   
95        1  02-12-2011    1584083.95             0        48.91       3.172   
...     ...         ...           ...           ...          ...         ...   
6340     45  07-01-2011     680254.35             0        34.32       3.193   
6341     45  14-01-2011     654018.95             0        24.78       3.205   
6342     45  21-01-2011     644285.33             0        30.55       3.229   
6362     45  10-06-2011     814395.17             0        73.88       3.881   
6434     45  26-10-2012     760281.43             0        58.85       3.882   

             CPI  Unemployment  
0     211.096358         8.106  
92    217.998085         7.866  
93    218.220509         7.866  
94    218.467621         7.866  
95    218.714733         7.866  
...          ...           ...  
6340  182.598306         8.549  
6341  182.658578         8.549  
6342  182.919337         8.549  
6362  185.648692         8.521  
6434  192.308899         8.667  

[6435 rows x 8 columns]

### 2.2.3 - Renomeando os índices
> Para maior facilidade na maniulação dos dados estarei passando os indices para letras miusculas.

In [418]:
df.rename(columns=str.lower, inplace=True)
pd.DataFrame.head(df)

store        date  weekly_sales  holiday_flag  temperature  fuel_price  \
0       1  05-02-2010    1643690.90             0        42.31       2.572   
92      1  11-11-2011    1594938.89             0        59.11       3.297   
93      1  18-11-2011    1539483.70             0        62.25       3.308   
94      1  25-11-2011    2033320.66             1        60.14       3.236   
95      1  02-12-2011    1584083.95             0        48.91       3.172   

           cpi  unemployment  
0   211.096358         8.106  
92  217.998085         7.866  
93  218.220509         7.866  
94  218.467621         7.866  
95  218.714733         7.866

## 3 - Análise dos dados

### 3.1 - Qual o preço médio de vendas semanal por loja?

In [424]:
df.describe(include='all') 

store        date  weekly_sales  holiday_flag  temperature  \
count   6435.000000        6435  6.435000e+03   6435.000000  6435.000000   
unique          NaN         143           NaN           NaN          NaN   
top             NaN  05-02-2010           NaN           NaN          NaN   
freq            NaN          45           NaN           NaN          NaN   
mean      23.000000         NaN  1.046965e+06      0.069930    60.663782   
std       12.988182         NaN  5.643666e+05      0.255049    18.444933   
min        1.000000         NaN  2.099862e+05      0.000000    -2.060000   
25%       12.000000         NaN  5.533501e+05      0.000000    47.460000   
50%       23.000000         NaN  9.607460e+05      0.000000    62.670000   
75%       34.000000         NaN  1.420159e+06      0.000000    74.940000   
max       45.000000         NaN  3.818686e+06      1.000000   100.140000   

         fuel_price          cpi  unemployment  
count   6435.000000  6435.000000   6435.000000  
unique          NaN          NaN           NaN  
top             NaN          NaN           NaN  
freq            NaN          NaN           NaN  
mean       3.358607   171.578394      7.999151  
std        0.459020    39.356712      1.875885  
min        2.472000   126.064000      3.879000  
25%        2.933000   131.735000      6.891000  
50%        3.445000   182.616521      7.874000  
75%        3.735000   212.743293      8.622000  
max        4.468000   227.232807     14.313000

In [432]:
loja_1 = df[df['store'] == 1]['weekly_sales']
print(f'Média de vendas semanal da Loja_1: {loja_1.mean():.2f}')

Média de vendas semanal da Loja_1: $1555264.40


In [433]:
loja_2 = df[df['store'] == 2]['weekly_sales']
print(f'Média de vendas semanal da Loja_2: {loja_2.mean():.2f}')

Média de vendas semanal da Loja_2: $1925751.34


In [434]:
loja_3 = df[df['store'] == 3]['weekly_sales']
print(f'Média de vendas semanal da Loja_3: {loja_3.mean():.2f}')

Média de vendas semanal da Loja_3: 402704.44
